Manatee.Json/Pointer/JsonPointer.cs

```diff
@@ -128,12 +142,12 @@ public bool IsChildOf(JsonPointer pointer)
 
 		internal JsonPointer CleanAndClone()
 		{
-			return new JsonPointer(this.WhereNotNull()) {_usesHash = _usesHash};
+			return new JsonPointer(this.WhereNotNull().ToList(), _usesHash);
```

Comment:
The `List<T>` constructor will enumerate the collection before adding the items.  How does this help?